test/uk/gov/hmrc/gform/graph/processor/UserCtxEvaluatorProcessorTest.scala

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2019 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.gform.graph.processor
+
+import cats.Id
+import uk.gov.hmrc.auth.core.retrieve.OneTimeLogin
+import uk.gov.hmrc.auth.core.{ EnrolmentIdentifier, Enrolments, AffinityGroup => CoreAffinityGroup, Enrolment => CoreEnrolment }
+import uk.gov.hmrc.gform.Spec
+import uk.gov.hmrc.gform.auth.models.AuthenticatedRetrievals
+import uk.gov.hmrc.gform.graph.NonConvertible
+import uk.gov.hmrc.gform.sharedmodel.formtemplate.{ EnrolledIdentifier, EnrolmentAuth, HmrcEnrolmentModule, HmrcSimpleModule, IdentifierName, Never, ServiceId, ServiceName, UserCtx, UserField, AffinityGroup => FTAffinityGroup, Enrolment => EEnrolment }
+
+class UserCtxEvaluatorProcessorTest extends Spec {
+
+  forAll(userCtxTable) { (field, serviceId, authConfig, expected) =>
+    it should s"evaluate a user ctx with $field, $serviceId and $authConfig return $expected" in new UserCtxEvaluatorProcessor[
+      Id] {
+      processEvaluation(materialisedRetrievalsAgent, UserCtx(field.asInstanceOf[UserField]), authConfig) should be(
```

Comment:
`asInstanceOf` is unsafe, do we really need it?

Reply:
I agree totally and I hate it, however it didn't work without because of the Option + is testing code so.....